### PR TITLE
BEX-448 - adding dlq for invoice-fulfilment

### PR DESCRIPTION
**Pull request by mgrzebykuw**

### CODE REVIEW

**Review by sbuliarca — APPROVED** [review]

File: dev-aws/kafka-shared/customer-billing.tf [hunk 1]
```diff
@@ -23,8 +23,39 @@ module "invoice_fulfillment_producer" {
...
+  cert_common_name = "customer-billing/bills-total-api"
+}
+
+resource "kafka_topic" "invoice_fulfillment_deadletter" {
+  name               = "bex.internal.accountreadytobefulfilled_deadletter"
+  replication_factor = 3
+  partitions         = 10
+  config = {
+    # keep data for 7 days
+    "retention.ms" = "604800000"
```
Comment (espang-uw) [inline dev-aws/kafka-shared/customer-billing.tf:38]: Can you increase the retention a little bit? This topic might need manual intervention and a week might be too rushed.

File: dev-aws/kafka-shared/customer-billing.tf [hunk 2]
```diff
@@ -23,8 +23,39 @@ module "invoice_fulfillment_producer" {
...
+
+module "invoice_fulfillment_producer" {
+  source = "../../modules/producer"
+
+  topic = kafka_topic.invoice_fulfillment_deadletter.name
+
+  cert_common_name = "customer-billing/invoice-fulfillment"
+}
+
+module "invoice_fulfillment_consumer" {
```
Comment (espang-uw) [inline dev-aws/kafka-shared/customer-billing.tf:54]: Can you delete the consumer for now. We will only produce and don't consume from the invoice-fulfillment service at the moment. If we want to keep it we have to change the consumer_group name and I don't know to what

Comment (mgrzebykuw) [reply dev-aws/kafka-shared/customer-billing.tf]: yes, In my head I was repeating "retries retries retries" hence why copied the consumer for no reason 🤦

File: dev-aws/kafka-shared/customer-billing.tf [hunk 3]
```diff
@@ -23,8 +23,30 @@ module "invoice_fulfillment_producer" {
...
+    # keep data for 14 days
+    "retention.ms" = "1209600000"
+    # allow max 1 MB for a message
+    "max.message.bytes" = "1048576"
+    "compression.type"  = "zstd"
+    "cleanup.policy"    = "delete"
+  }
+}
+
+module "invoice_fulfillment_producer" {
```
Comment (espang-uw) [inline dev-aws/kafka-shared/customer-billing.tf:46]: This has the same name as the module in l.15. I don't know how this reacts to duplicated module names.

I don't know if we want to have 1 module per service and we can combine the two somehow or if we want 1 module per (service, topic).

@sbuliarca Do you have an opinion on that?

Comment (espang-uw) [reply dev-aws/kafka-shared/customer-billing.tf:46]: Looking at this (https://github.com/utilitywarehouse/kafka-cluster-config/blob/main/modules/producer/variables.tf#L2) we probably can only configure 1 topic per module.

Comment (sbuliarca) [reply dev-aws/kafka-shared/customer-billing.tf:46]: >This has the same name as the module in l.15. I don't know how this reacts to duplicated module names.

I don't think it allows using the same name.

>I don't know if we want to have 1 module per service and we can combine the two somehow or if we want 1 module per (service, topic).

Unfortunately the TF plugin we're using https://github.com/Mongey/terraform-provider-kafka doesn't allow us to combine then, so I'd say 1 module per (service, topic) is our option.
I'd name this something like `invoice_fulfillment_deadletter_producer`

Comment (espang-uw) [reply dev-aws/kafka-shared/customer-billing.tf]: Thanks

**Review by sbuliarca — APPROVED** [review]

**Pull request merged**